src/contracts/core/DelegationManager.sol

```diff
@@ -197,41 +222,55 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
      * @dev Reverts if the `staker` is already undelegated.
      */
-    function undelegate(address staker) external onlyWhenNotPaused(PAUSED_UNDELEGATION) returns (bytes32 withdrawalRoot) {
+    function undelegate(
+        address staker
+    ) external onlyWhenNotPaused(PAUSED_UNDELEGATION) returns (bytes32 withdrawalRoot) {
         require(isDelegated(staker), "DelegationManager.undelegate: staker must be delegated to undelegate");
         address operator = delegatedTo[staker];
         require(!isOperator(staker), "DelegationManager.undelegate: operators cannot be undelegated");
         require(staker != address(0), "DelegationManager.undelegate: cannot undelegate zero address");
         require(
             msg.sender == staker ||
-            msg.sender == operator ||
-            msg.sender == _operatorDetails[operator].delegationApprover,
+                msg.sender == operator ||
+                msg.sender == _operatorDetails[operator].delegationApprover,
             "DelegationManager.undelegate: caller cannot undelegate staker"
         );
-        
+
         // remove any shares from the delegation system that the staker currently has delegated, if necessary
         // force the staker into "undelegation limbo" in the EigenPodManager if necessary
         if (eigenPodManager.podOwnerHasActiveShares(staker)) {
             uint256 podShares = eigenPodManager.forceIntoUndelegationLimbo(staker, operator);
             // remove delegated shares from the operator
-            _decreaseOperatorShares({operator: operator, staker: staker, strategy: beaconChainETHStrategy, shares: podShares});
+            _decreaseOperatorShares({
+                operator: operator,
+                staker: staker,
+                strategy: beaconChainETHStrategy,
+                shares: podShares
+            });
         }
         // force-queue a withdrawal of all of the staker's shares from the StrategyManager, if necessary
         if (strategyManager.stakerStrategyListLength(staker) != 0) {
             IStrategy[] memory strategies;
             uint256[] memory strategyShares;
-            (strategies, strategyShares, withdrawalRoot)
-                = strategyManager.forceTotalWithdrawal(staker);
+            (strategies, strategyShares, withdrawalRoot) = strategyManager.forceTotalWithdrawal(staker);
 
             for (uint256 i = 0; i < strategies.length; ) {
-                _decreaseOperatorShares({operator: operator, staker: staker, strategy: strategies[i], shares: strategyShares[i]});
+                _decreaseOperatorShares({
+                    operator: operator,
+                    staker: staker,
+                    strategy: strategies[i],
+                    shares: strategyShares[i]
+                });
 
                 unchecked {
                     ++i;
                 }
             }
         }
 
+        // push the operator's new stake to the StakeRegistry
+        _pushOperatorStakeUpdate(operator);
```

Comment:
looks like this will already happen above, in the calls to `_decreaseOperatorShares` (if they happen), but seems OK to have this here for extra coverage 🤷 This is a testnet-only change / not intended to be gas optimized.

Reply:
nevermind, it's only in the external version of the function. we need this here; resolving the thread.